lib/services/notification-api-service.js

```diff
@@ -109,6 +109,147 @@ function NotificationApiServiceFactory(
+        Notification Type is "node" and action is "alerts"
+
+         {
+         "type": "node",
+         "action": "alerts",
+         "typeId": "58d94cec316779d4126be134",
+         "data": {
+             "Context": "context string",
+             "EventId": "8689",
+             "EventTimestamp": "2017-03-29T15:39:34-0500",
+             "EventType": "Alert",
+             "MemberId": "7e675c8e-127a-11e7-9fc8-64006ac35232",
+             "Message": "The coin cell battery in CMC 1 is not working.",
+             "MessageArgs": ["1"],
+             "MessageArgs@odata.count": 1,
+             "MessageId": "CMC8572",
+             "Severity": "Critical",
+             "sourceIpAddress": "10.240.19.130",
+             "nodeId": "58d94cec316779d4126be134",
+             "ChassisName": "PowerEdge R630",
+             "ServiceTag": "4666482",
+             "SN": "CN747515A80855"
+         },
+         "severity": "critical",
+         "version": "1.0",
+         "createdAt": "2017-03-29T19:44:12.972Z"
+         }
+         */
+        var amqpMessage = {};
+
+        return new Promise.resolve()
+            .then(function(){
+                return getAlert(req);
+            })
+            .then(function(alert){
+                return  updateBmcInfo(alert);
+            })
+            .then(function(data){
+                return updateDeviceInfo(data);
+            })
+            .then(function(data){
+                amqpMessage = data;
+                return eventsProtocol.publishExternalEvent(data)
+                    .then(function () {
+                        return amqpMessage;
+                    });
+            });
+    };
+
+    function getAlert (req) {
+        var amqpMessage = {};
+        var ip = req.headers['x-forwarded-for'] || req.connection.remoteAddress ||
+            req.socket.remoteAddress || req.connection.socket.remoteAddress;
+        logger.info("Alert was received from the node with IP: " + ip);
+        var alert = _.defaults(req.swagger.query || {},req.query || {}, req.body || {});
+        if(typeof(alert) === 'object' &&  Object.keys(alert).length > 0) {
+            //Format the amqp message by adopting the rackHD notification standard
+            amqpMessage.type = "node";
+            amqpMessage.action = "alerts";
+            amqpMessage.data = alert;
+            amqpMessage.data.sourceIpAddress = ip;
+            if (alert.hasOwnProperty("Severity")) {
+                amqpMessage.severity = alert.Severity.toLowerCase();
+            }
+
+            return amqpMessage;
+        }
+        else{
+            //Unexpected decoded "x-www-forum-urlencoded" data
+            throw new Errors.BadRequestError(
+                'Unexpected alert data'
+            );
+        }
+    }
+
+    function updateBmcInfo(amqpMessage) {
+        var ip = amqpMessage.data.sourceIpAddress;
+
+        return waterline.catalogs.findMostRecent({ "data.IP Addresss": ip, source: "bmc"})
```

Comment:
I think the obms settings should take precedent because we should first trust user's setting, then turn to lookup in raw catalog. The order to find the BMC  mac address is:
- If the `host` in ipmi-obm-service is mac address, then use it directly.
- Then if the `host` in ipmi-obm-service is IP address, then check the lookup to find corresponding macAddress, if exists, then use it.
- Check the raw catalog.